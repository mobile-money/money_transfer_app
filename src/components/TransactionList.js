import React from "react";
import PropTypes from "prop-types";
import Transaction from "./Transaction";

const TransactionList = props => {
  TransactionList.propTypes = {
    transactions: PropTypes.instanceOf(Array)
  };
  return (
    <div className="transactions">
      <h2>History</h2>
      {props.transactions.map((transaction, index) => (
        <Transaction key={index} transaction={transaction} />
      ))}
    </div>
  );
};

export default TransactionList;
