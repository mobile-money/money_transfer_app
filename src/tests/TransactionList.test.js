import React from "react";
import renderer from "react-test-renderer";
import TransactionList from "../components/TransactionList";

const transactions = [
  {
    selectedName: "Mark",
    transferAmount: 12.5,
    newBalance: 325,
    date: "Feb 7",
    note: "lunch"
  },
  {
    selectedName: "Luka",
    transferAmount: 25,
    newBalance: 337.5,
    date: "Jan 20",
    note: "cab share"
  },
  {
    selectedName: "Josh",
    transferAmount: 10.5,
    newBalance: 327,
    date: "Jan 18",
    note: "cinema"
  }
];

it("renders correctly", () => {
  const tree = renderer
    .create(<TransactionList transactions={transactions} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
