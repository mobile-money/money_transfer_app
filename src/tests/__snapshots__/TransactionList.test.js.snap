// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="transactions"
>
  <h2>
    History
  </h2>
  <div
    className="transaction"
  >
    <div
      className="transaction__left__left"
    >
      <div
        className="transaction__left__message"
      >
        Sent €
        12.5
         to 
        Mark
      </div>
      <div
        className="transaction__left__note"
      >
        lunch
      </div>
    </div>
    <div
      className="transaction__right"
    >
      <div
        className="transaction__right__date"
      >
        Feb 7
      </div>
      <div
        className="transaction__right__balance"
      >
         
        Balance: € 
        325
      </div>
    </div>
  </div>
  <div
    className="transaction"
  >
    <div
      className="transaction__left__left"
    >
      <div
        className="transaction__left__message"
      >
        Sent €
        25
         to 
        Luka
      </div>
      <div
        className="transaction__left__note"
      >
        cab share
      </div>
    </div>
    <div
      className="transaction__right"
    >
      <div
        className="transaction__right__date"
      >
        Jan 20
      </div>
      <div
        className="transaction__right__balance"
      >
         
        Balance: € 
        337.5
      </div>
    </div>
  </div>
  <div
    className="transaction"
  >
    <div
      className="transaction__left__left"
    >
      <div
        className="transaction__left__message"
      >
        Sent €
        10.5
         to 
        Josh
      </div>
      <div
        className="transaction__left__note"
      >
        cinema
      </div>
    </div>
    <div
      className="transaction__right"
    >
      <div
        className="transaction__right__date"
      >
        Jan 18
      </div>
      <div
        className="transaction__right__balance"
      >
         
        Balance: € 
        327
      </div>
    </div>
  </div>
</div>
`;
